zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java

```diff
@@ -323,11 +325,33 @@ private void loadTable()
 		
 		if (fReportCube.getValue() !=null)
 			pa_ReportCube_ID = (Integer)fReportCube.getValue();
-		
-		//pa_ReportCube_ID = 1000000;		
-		if (pa_ReportCube_ID == 0 || c_PeriodTo_ID == 0 || m_AD_Org_ID == 0)
-		{
-			FDialog.error(form.getWindowNo(), "Please fill mandatory fields (Cube, Period, Organization).");
+
+		//pa_ReportCube_ID = 1000000;
+
+		MAcctSchema acctSchema = MAcctSchema.getClientAcctSchema(Env.getCtx(), Env.getAD_Client_ID(Env.getCtx()))[0];
+		if (acctSchema == null)
+			FDialog.error(form.getWindowNo(), "@No Acctschema@");
+		MAcctSchemaElement acctElementOrg = acctSchema.getAcctSchemaElement(MAcctSchemaElement.ELEMENTTYPE_Organization);
+		Boolean orgMandatory = acctElementOrg.isBalanced()?true:false;
+
+		StringBuffer errorMessage = new StringBuffer();
+		if (pa_ReportCube_ID == 0) {
+			errorMessage.append("@PA_ReportCube_ID@");
+		}
+		if (c_PeriodTo_ID == 0) {
+			if(errorMessage.length() > 0) {
+				errorMessage.append(Env.NL);
+			}
+			errorMessage.append("@C_PeriodTo_ID@");
+		}
+		if (orgMandatory && m_AD_Org_ID == 0) {
+			if(errorMessage.length() > 0) {
+				errorMessage.append(Env.NL);
+			}
+			errorMessage.append("@AD_Org_ID@");
+		}
+		if(errorMessage.length() > 0) {
+			FDialog.error(form.getWindowNo(), "FillMandatory", errorMessage.toString());

```

Comment:
Change for:
<pre>
FDialog.error(form.getWindowNo(), "FillMandatory", Msg.parseTranslation(Env.getCtx(), errorMessage.toString()));
</pre>